wrapanapi/systems/nuage.py

```diff
@@ -58,26 +70,191 @@ def common_headers(self):
             'X-Nuage-Organization': 'csp'
         }
 
+    #
+    # Integration-test specific entities on Nuage server
+    #
+
+    @property
+    def enterprise(self):
+        if not self._enterprise:
+            match = self.list_enterprises(name=self.enterprise_name)
+            if match:
+                self._enterprise = match[0]['ID']
+        return self._enterprise
+
+    @property
+    def domain_template(self):
+        if not self._domain_template:
+            match = self.list_domain_templates(name=self.domain_template_name)
+            if match:
+                self._domain_template = match[0]['ID']
+        return self._domain_template
+
+    @property
+    def domain(self):
+        if not self._domain:
+            match = self.list_domains_for_enterprise(name=self.domain_name)
+            if match:
+                self._domain = match[0]['ID']
+        return self._domain
+
+    @property
+    def zone(self):
+        if not self._zone:
+            match = self.list_zones_for_domain(name=self.zone_name)
+            if match:
+                self._zone = match[0]['ID']
+        return self._zone
+
+    @property
+    def subnet(self):
+        if not self._subnet:
+            match = self.list_cloud_subnets(name=self.subnet_name)
+            if match:
+                self._subnet = match[0]['ID']
+        return self._subnet
+
+    @enterprise.setter
+    def enterprise(self, value):
+        self._enterprise = value
+
+    #
+    # Multi API calls wrappers
+    #
+
+    def destroy_enterprise(self, name=None):
```

Comment:
Please use `cleanup_enterprise` to be consistent with current implementations of these types of methods that delete the target entity (enterprise) and all child resources.